biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/report/MultisampleReportBuilder.scala

```diff
@@ -25,36 +25,33 @@ import scala.concurrent.duration.Duration
 trait MultisampleReportBuilder extends ReportBuilder {
 
   /** Method to generate a single sample page */
-  def samplePage(sampleId: Int, args: Map[String, Any]): ReportPage
+  def samplePage(sampleId: Int, args: Map[String, Any]): Future[ReportPage]
 
   /** Default list of samples, can be override */
   def samplesSections: List[(String, ReportSection)] = {
-    List(
-      ("Samples", ReportSection("/nl/lumc/sasc/biopet/core/report/samplesList.ssp"))
-    )
+    List("Samples" -> ReportSection("/nl/lumc/sasc/biopet/core/report/samplesList.ssp"))
   }
 
   /** Method to generate a single library page */
-  def libraryPage(sampleId: Int, libraryId: Int, args: Map[String, Any]): ReportPage
+  def libraryPage(sampleId: Int, libraryId: Int, args: Map[String, Any]): Future[ReportPage]
 
   /** Default list of libraries, can be override */
   def librariesSections: List[(String, ReportSection)] = {
-    List(
-      ("Libraries", ReportSection("/nl/lumc/sasc/biopet/core/report/librariesList.ssp"))
-    )
+    List("Libraries" -> ReportSection("/nl/lumc/sasc/biopet/core/report/librariesList.ssp"))
   }
 
   /** Generate the samples page including a single sample page for each sample in the summary */
-  def generateSamplesPage(args: Map[String, Any]): ReportPage = {
+  def generateSamplesPage(args: Map[String, Any]): Future[ReportPage] = Future {
```

Comment:
Ah, but that's two different thing here. You can still make the report to the `subPages` non-blocking, right?

```
// Make the subpage `Future`s here
val subpages: Future[Seq[ReportPage]] = Future { ... }
// And map the results into a `ReportPage` here
val page = subpages.map { ... }
```